### PR TITLE
Foreman I18N support

### PR DESCRIPTION
**Pull request by lzap**
This one supersedes https://github.com/theforeman/foreman/pull/495

Rebased version with few more fixes by Dominic. Thanks all.

### CODE REVIEW

File: config/locales/update.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
+I18N_BRANCH="rails-3-x"
```
Comment (ohadlevy) [inline config/locales/update.sh]: whats the purpose of this file? should we change this branch to develop?

Comment (ohadlevy) [inline config/locales/update.sh]: oh, i c, not our branch - thanks

File: config/locales/show-missing.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
```
Comment (ohadlevy) [inline config/locales/show-missing.sh]: should we put these scripts under extras instead?

**Comment by ohadlevy** [comment]
adding all of the labels (were there was no label) is really not required, a simple change in helper:

``` diff
diff --git a/app/helpers/layout_helper.rb b/app/helpers/layout_helper.rb
index 7be665c..aa65119 100644
--- a/app/helpers/layout_helper.rb
+++ b/app/helpers/layout_helper.rb
@@ -121,7 +121,8 @@ module LayoutHelper
     help_inline = inline.blank? ? '' : content_tag(:span, inline, :class => "help-inline")
     help_block  = content_tag(:span, options.delete(:help_block), :class => "help-block")
     content_tag :div, :class => "control-group #{fluid ? "row-fluid" : ""} #{error.empty? ? "" : 'error'}" do
-      label_tag(attr, options.delete(:label), :class=>"control-label").html_safe +
+      label = options.delete(:label) || _(attr.to_s.humanize)
+      label_tag(attr, label, :class=>"control-label").html_safe +
         content_tag(:div, :class => "controls") do
           yield.html_safe + help_inline.html_safe + help_block.html_safe
         end.html_safe
```

I think it makes sense to undo that as it reduces the amount of changes required (plus it catches a few missing translations)

**Comment by ohadlevy** [comment]
@lzap another questions about the exception codes, currently does it mean that if we change the error message(in english) , you would get a different code? how about setting the code from where we are raising the exception? (optional for later usage)

**Comment by domcleal** [comment]
@ohadlevy the code is calculated using the untranslated string in `@message`, it's only translated on render.

**Comment by lzap** [comment]
1) I will move the scripts

2) that label - hmmm (didn't make my day today :-)

3) yes changing the error message changes the code. well, setting it dynamically based on stacktrace has one big disadvantage - I am about to write a simple tool that will collect all the exceptions from the codebase creating a simple table with all the codes and messages. this wont be possible with stacktraces.

Edit: As Dominic said, we only based the code on the exception class name and base (english) message

**Comment by lzap** [comment]
Labels are being extracted in the following format: https://github.com/lzap/foreman/blob/gettext-rebased/locale/model_attributes.rb

We need that because this is what gettext_i18n_rails plugin does - it is used in error messages. Now, to leverage that and get rid of :labels we need the model context which we don't have in erb - we only have HTML form object and field symbol (e.g. :name) but we need model itself and attribute name/symbol.

The gettext_i18n_rails plugin code that converts the column name into gettext string is here: https://github.com/grosser/gettext_i18n_rails/blob/master/lib/gettext_i18n_rails/active_model/translation.rb

**Comment by lzap** [comment]
Closing because: https://github.com/theforeman/foreman/pull/545

**Pull request closed without merging**